pkg/api/v1/node/util_test.go

```diff
@@ -0,0 +1,133 @@
+/*
+Copyright 2015 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package node
+
+import (
+	"github.com/stretchr/testify/assert"
+	"k8s.io/api/core/v1"
+	"testing"
+)
+
+func TestGetNodeCondition(t *testing.T) {
+	nodeStatus := v1.NodeStatus{
+		Conditions: []v1.NodeCondition{
+			{
+				Type:   v1.NodeReady,
+				Status: v1.ConditionTrue,
+			},
+			{
+				Type:   v1.NodeOutOfDisk,
+				Status: v1.ConditionFalse,
+			},
+			{
+				Type:   v1.NodeMemoryPressure,
+				Status: v1.ConditionFalse,
+			},
+		},
+	}
+	tests := []struct {
+		status        *v1.NodeStatus
+		condition     v1.NodeCondition
+		expectedIndex int
+	}{
+		{
+			status: &nodeStatus,
+			condition: v1.NodeCondition{
+				Type:   v1.NodeReady,
+				Status: v1.ConditionTrue},
+			expectedIndex: 0,
+		},
+		{
+			status: &nodeStatus,
+			condition: v1.NodeCondition{
+				Type:   v1.NodeOutOfDisk,
+				Status: v1.ConditionFalse},
+			expectedIndex: 1,
+		},
+		{
+			status: &nodeStatus,
+			condition: v1.NodeCondition{
+				Type:   v1.NodeMemoryPressure,
+				Status: v1.ConditionFalse},
+			expectedIndex: 2,
+		},
+		{
+			status: &nodeStatus,
+			condition: v1.NodeCondition{
+				Type:   v1.NodeDiskPressure,
+				Status: v1.ConditionFalse},
+			expectedIndex: -1,
+		},
+	}
+
+	for _, test := range tests {
+		index, condition := GetNodeCondition(test.status, test.condition.Type)
+
+		assert.Equal(t, test.expectedIndex, index, "unexpected result of test: %v, expected index: %v, actual index: %v", test, test.expectedIndex, index)
```

Comment:
1. name your test cases, rather then print out the entire `test` in the error message.
2. using the `assert` library is making the line long and less readable, how about simply
```golang 
if test.expectedIndex != index {
  t.Errorf("%q: expected %v, got %v", test.name, test.expectedIndex, index)
}
```
Same comment for the other uses of `assert` in this pull.